### DISCUSSION TITLE
Does riverpod generator support generic functions?

### DISCUSSION THREAD

**Discussion by MkamaMakoko** [rrousselGit/riverpod#1968] [Q&A] [answered]
I was trying to generate a provider from a generic function, the file generation completed, but the generated file had some errors

My function is like this

```
@riverpod
MyObject<T> object<T>(ObjectRef ref, {required T item}){
...
}
```

I'm I doing something wrong, or generator doesn't support generic functions?

Generator for generic classes of (Async)Notifier provider works fines.

Thanks

**Comment by rrousselGit** [comment, marked as answer]
No, this isn't supported.
